app/controllers/register/trustees/individual/TrusteeNinoNewController.scala

```diff
@@ -45,40 +46,30 @@ class TrusteeNinoNewController @Inject()(
                                  )(implicit val ec: ExecutionContext) extends NinoController with I18nSupport {
 
   private[controllers] val postCall = controllers.register.trustees.individual.routes.TrusteeNinoNewController.onSubmit _
-  private[controllers] val title: Message = "messages__common_nino__title"
-  private[controllers] val heading: Message = "messages__common_nino__h1"
-  private[controllers] val hint: Message = "messages__common__nino_hint"
 
-  private def viewmodel(index: Index,  mode: Mode, srn: Option[String]): Retrieval[NinoViewModel] =
-    Retrieval {
-      implicit request =>
-        TrusteeDetailsId(index).retrieve.right.map {
-          details =>
-            NinoViewModel(
-              postCall(mode, Index(index), srn),
-              title = title,
-              heading = heading,
-              hint = hint,
-              personName = details.fullName,
-              srn = srn
-            )
-        }
-    }
+  private def viewmodel(personDetails: PersonDetails, index: Index,  mode: Mode, srn: Option[String]): NinoViewModel =
+    NinoViewModel(
+      postCall(mode, Index(index), srn),
+      title = Message("messages__trustee__individual__nino__title"),
+      heading = Message("messages__trustee__individual__nino__heading", personDetails.firstAndLastName),
+      hint = Message("messages__common__nino_hint"),
+      srn = srn
+    )
 
   def onPageLoad(mode: Mode, index: Index, srn: Option[String]): Action[AnyContent] =
     (authenticate andThen getData(mode, srn) andThen allowAccess(srn) andThen requireData).async {
     implicit request =>
```

Comment:
Use TrusteeNameId when the toggle is on. This test is bound to break once the toggle is switched on and will need to be fixed at the last moment

Reply:
Addressed as suggested, but this was missed when we make breaking changes to the cache. We might need to have a closer look at this issue since there may be more